blueprints/pre0.21.8-supabase/template.toml

```
@@ -0,0 +1,1069 @@
+content = """import { serve } from 'https://deno.land/std@0.131.0/http/server.ts'
+import * as jose from 'https://deno.land/x/jose@v4.14.4/index.ts'
+
+console.log('main function started')
+
+const JWT_SECRET = Deno.env.get('JWT_SECRET')
+const VERIFY_JWT = Deno.env.get('VERIFY_JWT') === 'true'
+
+function getAuthToken(req: Request) {
+  const authHeader = req.headers.get('authorization')
+  if (!authHeader) {
+    throw new Error('Missing authorization header')
+  }
+  const [bearer, token] = authHeader.split(' ')
+  if (bearer !== 'Bearer') {
+    throw new Error(`Auth header is not 'Bearer {token}'`)
+  }
+  return token
+}
+
+async function verifyJWT(jwt: string): Promise<boolean> {
+  const encoder = new TextEncoder()
+  const secretKey = encoder.encode(JWT_SECRET)
+  try {
+    await jose.jwtVerify(jwt, secretKey)
+  } catch (err) {
+    console.error(err)
+    return false
+  }
+  return true
+}
+
+serve(async (req: Request) => {
+  if (req.method !== 'OPTIONS' && VERIFY_JWT) {
+    try {
+      const token = getAuthToken(req)
+      const isValidJWT = await verifyJWT(token)
+
+      if (!isValidJWT) {
+        return new Response(JSON.stringify({ msg: 'Invalid JWT' }), {
+          status: 401,
+          headers: { 'Content-Type': 'application/json' },
+        })
+      }
+    } catch (e) {
+      console.error(e)
+      return new Response(JSON.stringify({ msg: e.toString() }), {
+        status: 401,
+        headers: { 'Content-Type': 'application/json' },
+      })
+    }
+  }
+
+  const url = new URL(req.url)
+  const { pathname } = url
+  const path_parts = pathname.split('/')
+  const service_name = path_parts[1]
+
+  if (!service_name || service_name === '') {
+    const error = { msg: 'missing function name in request' }
+    return new Response(JSON.stringify(error), {
+      status: 400,
+      headers: { 'Content-Type': 'application/json' },
+    })
+  }
+
+  const servicePath = `/home/deno/functions/${service_name}`
+  console.error(`serving the request with ${servicePath}`)
+
+  const memoryLimitMb = 150
+  const workerTimeoutMs = 1 * 60 * 1000
+  const noModuleCache = false
+  const importMapPath = null
+  const envVarsObj = Deno.env.toObject()
+  const envVars = Object.keys(envVarsObj).map((k) => [k, envVarsObj[k]])
+
+  try {
+    const worker = await EdgeRuntime.userWorkers.create({
+      servicePath,
+      memoryLimitMb,
+      workerTimeoutMs,
+      noModuleCache,
+      importMapPath,
+      envVars,
+    })
+    return await worker.fetch(req)
+  } catch (e) {
+    const error = { msg: e.toString() }
+    return new Response(JSON.stringify(error), {
+      status: 500,
+      headers: { 'Content-Type': 'application/json' },
+    })
+  }
+})
+"""
+
+[[config.mounts]]
+filePath = "/volumes/logs/vector.yml"
+content = """api:
+  enabled: true
+  address: 0.0.0.0:9001
+
+sources:
+  docker_host:
+    type: docker_logs
+    exclude_containers:
+      - ${container_name_prefix}-vector
```

I mean but having the appName, we can easily know which container it refers to within each volume for example something like this 

```suggestion
      - ${APP_NAME}-vector_1
```

this should work, if I'm not wrong

---

We also need more than that: for logs to appear in supabase they require the correct .appname in the vector.yml
```
      kong: '.appname == "supabase-kong"'
      auth: '.appname == "supabase-auth"'
      rest: '.appname == "supabase-rest"'
      realtime: '.appname == "realtime-dev.supabase-realtime"'
      storage: '.appname == "supabase-storage"'
      functions: '.appname == "supabase-edge-functions"'
      db: '.appname == "supabase-db"'
```
first because they are filtered by those names, but also because inside supabase studio they won't display anything with a different name than that even if you changes this settings to the correct container name. 

So to make it work I've added a replace : ```.appname = replace!(del(.container_name), "${container_name_prefix}", "supabase")``` (wich i'm not fan of as it adds small overhead on each log received)

This also mean we should not add suffix like in your example: ```${APP_NAME}-vector_1``` unless we also remove it. 
So we potentially can replace this with something like: ```.appname = replace!(del(.container_name), "${APP_NAME}", "supabase")``` but in the end it is mandatory for logs to appear in supabase studio to use this names in the log themselves (ie: supabase-kong, supabase-auth, supabase-rest...).

I really think that the most simple solution is to simply change the ```container_name_prefix``` to include the ```${APP_NAME}``` as proposed earlier.

---

@Siumauricio we can perhaps just reverse the prefix like this: 
```
[variables]
main_domain = "${domain}"
...
container_name_prefix = "${appName}-supabase" 
```
This will keep more meaningful names, and I think it is easy to lookup in the env file the CONTAINER_PREFIX  value to see correlation within the docker-compose.yml

---

Yes @malko, please, with those changes I merge! sorry for the delay!